client/src/utils/pg/build.ts

```diff
@@ -13,17 +14,51 @@ interface BuildResp {
 export type Files = string[][];
 
 export class PgBuild {
+  static getProgramName(fileName: string, expectedSuffix: string) {
+    const pathParts = fileName.split("/");
+    const lastPart = pathParts[pathParts.length - 1];
+    return lastPart.replace(expectedSuffix, "");
+  }
+
   static async build(files: Files) {
     const programInfo = PgProgramInfo.getProgramInfo();
 
+    const pythonFiles = files.filter(([fileName]) =>
+      fileName.toLowerCase().endsWith(".py")
+    );
+
+    let rustFiles: Files;
+    if (pythonFiles.length > 0) {
+      const seahorsePkg = await PgPkg.loadPkg(PkgName.SEAHORSE_COMPILE);
```

Comment:
This line will re-import the seahorse wasm crate every time it runs. Would be nice to have some sort of in-memory access to this function after the first import similar to https://github.com/solana-playground/solana-playground/blob/master/client/src/components/Panels/Main/Terminal/usePkg.tsx#L27

Reply:
Nice, I've refactored this a bit. The package state is now maintained in `useBuild`, and the logic to determine if we have Python files is there too. So the package will only be loaded the first time a build with Python files happens.

I've also refactored in this file to a `buildPython` and `buildRust` function, where the `buildPython` ones takes the package as input, compiles to rust and then calls/returns `buildRust`. I think that's a bit cleaner too, it removes any package loading from here and better shows how the Python and Rust builds relate to each other.

Reply:
Yes, it's much better now!